### PR TITLE
Only require exit confirmation when there are background operations

### PR DESCRIPTION
**Pull request by peppy**
- When exiting to install an update, the user is not prompted for confirmation unless there are ongoing operations
- When performing an operation that requires a restart (ie. changing rendrer) on a platform that supports it (ie. windows) the game will no longer confirm and will automatically restart as one would expect.
- When there are ongoing operations, a dangerous confirmation is required, and a list of the ongoing operations is shown in the confirmation dialog.

https://github.com/ppy/osu/assets/191335/e1884846-24ea-4ce6-9dd0-92e585860fac

### CODE REVIEW

**Comment by peppy** [comment]
I think this is in a pretty good place now.

File: osu.Game/Screens/Menu/ConfirmExitDialog.cs [hunk 1]
```diff
@@ -2,38 +2,79 @@
...
-                },
-                new PopupDialogCancelButton
+                    new PopupDialogDangerousButton
+                    {
+                        Text = @"Let me out!",
+                        Action = onConfirm
+                    },
+                    new PopupDialogCancelButton
+                    {
+                        Text = @"Cancel",
```
Comment (soopax) [inline osu.Game/Screens/Menu/ConfirmExitDialog.cs:55]: This is kind of confusing. Cancel the exit process or cancel the ongoing tasks? I suggest that this should say "Back" instead.

**Review by soopax — CHANGES_REQUESTED** [review]

File: osu.Game/Overlays/NotificationOverlay.cs [hunk 2]
```diff
@@ -33,6 +34,9 @@ public partial class NotificationOverlay : OsuFocusedOverlayContainer, INamedOve
 
         public const float TRANSITION_LENGTH = 600;
 
+        public IEnumerable<Notification> AllNotifications =>
+            toastTray.Notifications.Concat(sections.SelectMany(s => s.Notifications));
```
Comment (bdach) [inline osu.Game/Overlays/NotificationOverlay.cs:38]: Not sure why the tests are failing on this and I can't reproduce locally but it does look a touch unsafe to access before loaded... Maybe this should say

```suggestion
        public IEnumerable<Notification> AllNotifications =>
            IsLoaded ? toastTray.Notifications.Concat(sections.SelectMany(s => s.Notifications)) : Array.Empty<Notification>();
```

or something.

File: osu.Game.Tests/Visual/UserInterface/TestSceneHoldToConfirmOverlay.cs
Comment (bdach) [inline osu.Game.Tests/Visual/UserInterface/TestSceneHoldToConfirmOverlay.cs:1]: Test file needs renaming to match new class name.

File: osu.Game/Screens/Menu/ConfirmExitDialog.cs (hunk 1 above)
Comment (bdach) [reply osu.Game/Screens/Menu/ConfirmExitDialog.cs:55]: Agree with this.

**Review by bdach — APPROVED** [review]
Have tested what I can (the background operation confirms). Cannot test the autorestart functionality or the assertion that the game will autorestart without confirmation without a proper release soooo.... probably just gonna trust that it works as it should

**Pull request merged**